Proper exception message when passing null value to SyntheticBeanBuildItem Configuration parameter

### Description

When you pass a null value as parameter to the SyntheticBeanBuildItem generation, you get a NullPointerException but you don't have any clue (at first) where the problem occured.

```
Caused by: java.lang.NullPointerException
        at io.quarkus.gizmo.BytecodeCreatorImpl.allocateLocalVariables(BytecodeCreatorImpl.java:1195)
        at io.quarkus.gizmo.MethodCreatorImpl.write(MethodCreatorImpl.java:122)
        at io.quarkus.gizmo.ClassCreator.writeTo(ClassCreator.java:202)
        at io.quarkus.gizmo.ClassCreator.close(ClassCreator.java:225)
        at io.quarkus.arc.processor.BeanGenerator.generateSyntheticBean(BeanGenerator.java:222)

```
You can trigger the above Exception by code like this

```
        return SyntheticBeanBuildItem.configure(Foo.class)
                .scope(Singleton.class)
                .creator(FooCreator.class)
                .param("test", (String)null)
                .done();
```

Of course, in our extension the parameter test was based on some logic and it was not clear that the NPE was caused by passing `null` as value as it happens later and stacktrace is not related anymore to the problematic statement.



### Implementation ideas

Add a proper check and exception to all `.param` overloaded methods to indicate null is not an allowed value.

--- a/independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/ConfiguratorBase.java
+++ b/independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/ConfiguratorBase.java
@@ -2,6 +2,7 @@ package io.quarkus.arc.processor;
 
 import java.util.HashMap;
 import java.util.Map;
+import java.util.Objects;
 import org.jboss.jandex.AnnotationInstance;
 import org.jboss.jandex.ClassInfo;
 
@@ -31,6 +32,7 @@ public abstract class ConfiguratorBase<THIS extends ConfiguratorBase<THIS>> {
     }
 
     public THIS param(String name, boolean[] value) {
+        Objects.requireNonNull(value, "Parameter value can't be null");
         params.put(name, value);
         return self();
     }
@@ -41,6 +43,7 @@ public abstract class ConfiguratorBase<THIS extends ConfiguratorBase<THIS>> {
     }
 
     public THIS param(String name, byte[] value) {
+        Objects.requireNonNull(value, "Parameter value can't be null");
         params.put(name, value);
         return self();
     }
@@ -51,6 +54,7 @@ public abstract class ConfiguratorBase<THIS extends ConfiguratorBase<THIS>> {
     }
 
     public THIS param(String name, short[] value) {
+        Objects.requireNonNull(value, "Parameter value can't be null");
         params.put(name, value);
         return self();
     }
@@ -61,6 +65,7 @@ public abstract class ConfiguratorBase<THIS extends ConfiguratorBase<THIS>> {
     }
 
     public THIS param(String name, int[] value) {
+        Objects.requireNonNull(value, "Parameter value can't be null");
         params.put(name, value);
         return self();
     }
@@ -71,6 +76,7 @@ public abstract class ConfiguratorBase<THIS extends ConfiguratorBase<THIS>> {
     }
 
     public THIS param(String name, long[] value) {
+        Objects.requireNonNull(value, "Parameter value can't be null");
         params.put(name, value);
         return self();
     }
@@ -81,6 +87,7 @@ public abstract class ConfiguratorBase<THIS extends ConfiguratorBase<THIS>> {
     }
 
     public THIS param(String name, float[] value) {
+        Objects.requireNonNull(value, "Parameter value can't be null");
         params.put(name, value);
         return self();
     }
@@ -91,6 +98,7 @@ public abstract class ConfiguratorBase<THIS extends ConfiguratorBase<THIS>> {
     }
 
     public THIS param(String name, double[] value) {
+        Objects.requireNonNull(value, "Parameter value can't be null");
         params.put(name, value);
         return self();
     }
@@ -101,56 +109,67 @@ public abstract class ConfiguratorBase<THIS extends ConfiguratorBase<THIS>> {
     }
 
     public THIS param(String name, char[] value) {
+        Objects.requireNonNull(value, "Parameter value can't be null");
         params.put(name, value);
         return self();
     }
 
     public THIS param(String name, String value) {
+        Objects.requireNonNull(value, "Parameter value can't be null");
         params.put(name, value);
         return self();
     }
 
     public THIS param(String name, String[] value) {
+        Objects.requireNonNull(value, "Parameter value can't be null");
         params.put(name, value);
         return self();
     }
 
     public THIS param(String name, Enum<?> value) {
+        Objects.requireNonNull(value, "Parameter value can't be null");
         params.put(name, value);
         return self();
     }
 
     public THIS param(String name, Enum<?>[] value) {
+        Objects.requireNonNull(value, "Parameter value can't be null");
         params.put(name, value);
         return self();
     }
 
     public THIS param(String name, Class<?> value) {
+        Objects.requireNonNull(value, "Parameter value can't be null");
         params.put(name, value);
         return self();
     }
 
     public THIS param(String name, Class<?>[] value) {
+        Objects.requireNonNull(value, "Parameter value can't be null");
         params.put(name, value);
         return self();
     }
 
     public THIS param(String name, ClassInfo value) {
+        Objects.requireNonNull(value, "Parameter value can't be null");
         params.put(name, value);
         return self();
     }
 
     public THIS param(String name, ClassInfo[] value) {
+        Objects.requireNonNull(value, "Parameter value can't be null");
         params.put(name, value);
         return self();
     }
 
     public THIS param(String name, AnnotationInstance value) {
+        Objects.requireNonNull(value, "Parameter value can't be null");
         params.put(name, value);
         return self();
     }
 
     public THIS param(String name, AnnotationInstance[] value) {
+        Objects.requireNonNull(value, "Parameter value can't be null");
         params.put(name, value);
         return self();
     }

--- a/independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/ConfiguratorBase.java
+++ b/independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/ConfiguratorBase.java
@@ -32,8 +32,7 @@ public abstract class ConfiguratorBase<THIS extends ConfiguratorBase<THIS>> {
     }
 
     public THIS param(String name, boolean[] value) {
-        Objects.requireNonNull(value, "Parameter value can't be null");
-        params.put(name, value);
+        params.put(name, Objects.requireNonNull(value));
         return self();
     }
 
@@ -43,8 +42,7 @@ public abstract class ConfiguratorBase<THIS extends ConfiguratorBase<THIS>> {
     }
 
     public THIS param(String name, byte[] value) {
-        Objects.requireNonNull(value, "Parameter value can't be null");
-        params.put(name, value);
+        params.put(name, Objects.requireNonNull(value));
         return self();
     }
 
@@ -54,8 +52,7 @@ public abstract class ConfiguratorBase<THIS extends ConfiguratorBase<THIS>> {
     }
 
     public THIS param(String name, short[] value) {
-        Objects.requireNonNull(value, "Parameter value can't be null");
-        params.put(name, value);
+        params.put(name, Objects.requireNonNull(value));
         return self();
     }
 
@@ -65,8 +62,7 @@ public abstract class ConfiguratorBase<THIS extends ConfiguratorBase<THIS>> {
     }
 
     public THIS param(String name, int[] value) {
-        Objects.requireNonNull(value, "Parameter value can't be null");
-        params.put(name, value);
+        params.put(name, Objects.requireNonNull(value));
         return self();
     }
 
@@ -76,8 +72,7 @@ public abstract class ConfiguratorBase<THIS extends ConfiguratorBase<THIS>> {
     }
 
     public THIS param(String name, long[] value) {
-        Objects.requireNonNull(value, "Parameter value can't be null");
-        params.put(name, value);
+        params.put(name, Objects.requireNonNull(value));
         return self();
     }
 
@@ -87,8 +82,7 @@ public abstract class ConfiguratorBase<THIS extends ConfiguratorBase<THIS>> {
     }
 
     public THIS param(String name, float[] value) {
-        Objects.requireNonNull(value, "Parameter value can't be null");
-        params.put(name, value);
+        params.put(name, Objects.requireNonNull(value));
         return self();
     }
 
@@ -98,8 +92,7 @@ public abstract class ConfiguratorBase<THIS extends ConfiguratorBase<THIS>> {
     }
 
     public THIS param(String name, double[] value) {
-        Objects.requireNonNull(value, "Parameter value can't be null");
-        params.put(name, value);
+        params.put(name, Objects.requireNonNull(value));
         return self();
     }
 
@@ -109,68 +102,57 @@ public abstract class ConfiguratorBase<THIS extends ConfiguratorBase<THIS>> {
     }
 
     public THIS param(String name, char[] value) {
-        Objects.requireNonNull(value, "Parameter value can't be null");
-        params.put(name, value);
+        params.put(name, Objects.requireNonNull(value));
         return self();
     }
 
     public THIS param(String name, String value) {
-        Objects.requireNonNull(value, "Parameter value can't be null");
-        params.put(name, value);
+        params.put(name, Objects.requireNonNull(value));
         return self();
     }
 
     public THIS param(String name, String[] value) {
-        Objects.requireNonNull(value, "Parameter value can't be null");
-        params.put(name, value);
+        params.put(name, Objects.requireNonNull(value));
         return self();
     }
 
     public THIS param(String name, Enum<?> value) {
-        Objects.requireNonNull(value, "Parameter value can't be null");
-        params.put(name, value);
+        params.put(name, Objects.requireNonNull(value));
         return self();
     }
 
     public THIS param(String name, Enum<?>[] value) {
-        Objects.requireNonNull(value, "Parameter value can't be null");
-        params.put(name, value);
+        params.put(name, Objects.requireNonNull(value));
         return self();
     }
 
     public THIS param(String name, Class<?> value) {
-        Objects.requireNonNull(value, "Parameter value can't be null");
-        params.put(name, value);
+        params.put(name, Objects.requireNonNull(value));
         return self();
     }
 
     public THIS param(String name, Class<?>[] value) {
-        Objects.requireNonNull(value, "Parameter value can't be null");
-        params.put(name, value);
+        params.put(name, Objects.requireNonNull(value));
         return self();
     }
 
     public THIS param(String name, ClassInfo value) {
-        Objects.requireNonNull(value, "Parameter value can't be null");
-        params.put(name, value);
+        params.put(name, Objects.requireNonNull(value));
         return self();
     }
 
     public THIS param(String name, ClassInfo[] value) {
-        Objects.requireNonNull(value, "Parameter value can't be null");
-        params.put(name, value);
+        params.put(name, Objects.requireNonNull(value));
         return self();
     }
 
     public THIS param(String name, AnnotationInstance value) {
-        Objects.requireNonNull(value, "Parameter value can't be null");
-        params.put(name, value);
+        params.put(name, Objects.requireNonNull(value));
         return self();
     }
 
     public THIS param(String name, AnnotationInstance[] value) {
-        Objects.requireNonNull(value, "Parameter value can't be null");
-        params.put(name, value);
+        params.put(name, Objects.requireNonNull(value));
         return self();
     }
 

